src/main/java/model/LadderRow.java

```diff
@@ -1,20 +1,34 @@
 package model;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.IntStream;
 
 public class LadderRow {
 
     private final List<Line> isLines;
 
-    public LadderRow(List<Boolean> isLines) {
-        validateLadderRow(isLines);
-        this.isLines = isLines.stream().map(Line::valueOf).toList();
+    public LadderRow(BooleanGenerator generator, int size) {
+        List<Boolean> row = new ArrayList<>();
+        row.add(generator.generate());
+        IntStream.range(1, size).forEach(index -> row.add(generateNext(row.get(index - 1), generator)));
+        this.isLines = row.stream().map(Line::valueOf).toList();
+    }
+
+    LadderRow(List<Boolean> row) {
+        validateLadderRow(row);
+        this.isLines = row.stream().map(Line::valueOf).toList();
     }
 
     private void validateLadderRow(List<Boolean> isLines) {
-        for (int i = 1; i < isLines.size(); i++) {
-            isConsecutiveTrue(isLines, i);
+        IntStream.range(1, isLines.size()).forEach(index -> isConsecutiveTrue(isLines, index));
```

Comment:
한줄에 점 하나씩!
또한, stream forEach는 종단연산에 해당되기 때문에 현재와 같은 경우에는 사용을 지양해야 합니다.
stream for와 for loop의 차이에 대해 알아봅시다 🙂
https://tecoble.techcourse.co.kr/post/2020-05-14-foreach-vs-forloop/
```suggestion
        IntStream.range(1, isLines.size())
            .forEach(index -> isConsecutiveTrue(isLines, index));
```

Reply:
네 감사합니다.  해당 부분은 reduce를 사용하여 변경했으나, LadderRow에서 이미 중복이 되지 않도록 만들기 때문에 연속된 true 값이 있는지 검증하는 로직이 필요 없습니다. 해당 내용은 삭제했습니다!